### ISSUE TITLE
Combine csharp and visualbasic rule for "prefer_inferred_tuple_names" to just "dotnet"

### ISSUE THREAD

**Issue by kuhlenh** [dotnet/roslyn#23313]
We have the following rules:

```
csharp_prefer_inferred_tuple_names = true:suggestion 
csharp_prefer_inferred_anonymous_type_member_names = true:suggestion 
visual_basic_prefer_inferred_tuple_names = true:suggestion 
visual_basic_prefer_inferred_anonymous_type_member_names = true:suggestion
```

Is there a reason we chose to split C# and VB into separate rules as opposed to having a single `dotnet_prefer_inferred_tuple_names`  and `dotnet_prefer_inferred_tuple_names` rule?

**Comment by kuhlenh** [comment]
@CyrusNajmabadi -- do you remember why you had these as separate by language?

**Comment by CyrusNajmabadi** [comment]
I believe @jcouv wrote that feature.  Likely he copied from something else and no one caught it during review.

**Comment by kuhlenh** [comment]
@jcouv do you remember your thinking here? just wondering if I should hold off documenting this if we believe each pair of rules should be combined under a single "dotnet" rule...

**Comment by jcouv** [comment]
I don't have a preference either way. I either followed what was done for other preferences, or I was given some feedback to do per-language option.
I'm happy to merge them if you'd like. Let me know.

**Comment by kuhlenh** [comment]
@AnthonyDGreen do you feel a C# and VB developer would have different preferences for inferred tuple element naming and anonymous type member naming?

**Comment by kuhlenh** [comment]
Doh! @jcouv Just realized we definitely don't want this rule split out. Customers can have the single dotnet rule have different styles across languages by doing the following:

```
[*.cs]
dotnet_prefer_inferred_tuple_names = true:suggestion 
dotnet_prefer_inferred_anonymous_type_member_names = true:suggestion 

[*.vb]
dotnet_prefer_inferred_tuple_names = true:suggestion 
dotnet_prefer_inferred_anonymous_type_member_names = true:suggestion 
```

**Comment by jcouv** [comment]
@kuhlenh Thanks for the clarification. Does that mean we generally don't want any language-specific code style options (the filter solution you described should always work)?

@CyrusNajmabadi @kuhlenh I'll merge the options above. Do I need to include migration logic (still honoring old options) or can I implement a hard switch (delete the old options, implement new options instead)?

**Comment by kuhlenh** [comment]
@jcouv Yes, there should only be language-specific style options for rules that are language-specific (like rules with braces, local functions, etc.)

Also, I did not document/blog/announce these rules so we should be ok with deleting them entirely and making only the dotnet rules...but I will add @jinujoseph and @Pilchie to make sure they are ok with that plan.

**Comment by jcouv** [comment]
@kuhlenh I'm going to use "dotnet_style_prefer_inferred_tuple_names" (note `_style_`) rather than "dotnet_prefer_inferred_tuple_names" as that seems to be the naming convention for similar options.

**Comment by kuhlenh** [comment]
@jmarolf @dpoeschl can you confirm that the above name matches the pattern you all have been using?

**Comment by jmarolf** [comment]
yep `dotnet_style_name_of_options` is what we do for code styles

**Comment by jcouv** [comment]
Fixed in 15.6 (should be preview 2). Thanks
